src/libraries/System.Data.Odbc/src/Common/System/Data/Common/SafeNativeMethods.cs

```diff
@@ -1,28 +1,21 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 
 namespace System.Data
 {
     internal static partial class SafeNativeMethods
     {
-        internal static IntPtr LocalAlloc(nint initialSize)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe void ZeroMemory(IntPtr ptr, int length)
         {
-            var handle = Marshal.AllocHGlobal(initialSize);
-            ZeroMemory(handle, (int)initialSize);
-            return handle;
-        }
-
-        internal static void LocalFree(IntPtr ptr)
-        {
-            Marshal.FreeHGlobal(ptr);
-        }
-
-        internal static void ZeroMemory(IntPtr ptr, int length)
-        {
-            var zeroes = new byte[length];
-            Marshal.Copy(zeroes, 0, ptr, length);
+#if !NET7_0_OR_GREATER
+            new Span<byte>((void*)ptr, length).Clear();
```

Comment:
why not just Span.Clear for all TFMs ?

Reply:
Codegen is marginally better, see https://www.diffchecker.com/Gxzd78oe

The main reason - the `#if` indicates we could just use `NativeMemory.Clear` once .NET 6 is end-of-life.

Reply:
> Codegen is marginally better, see https://www.diffchecker.com/Gxzd78oe

It seems that you compared codegen for constant length which is not the case here. Overall, Span.Clear should be better e.g. it's faster for small sizes as it won't pay price for the interop machinery

Reply:
Codegen is not for constant length, see https://csharp.godbolt.org/z/a157vhTT9

Reply:
The implementation for `NativeMemory.Clear` actually uses `SpanHelpers.ClearWithoutReferences(ref *(byte*)ptr, byteCount)`

Reply:
Actually, since `NativeMemory.Alloc` guarantees `ptr` is pointer-aligned, and `Length` is always `IntPtr` aligned, we can use `Unsafe.InitBlock`.